src/main/java/com/google/sps/servlets/DeleteClass.java

```diff
@@ -0,0 +1,156 @@
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.sps.firebase.FirebaseAppManager;
+import com.google.sps.tasks.TaskSchedulerFactory;
+import com.google.sps.workspace.WorkspaceFactory;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/delete-class")
+public class DeleteClass extends HttpServlet {
+  private FirebaseAuth authInstance;
+  private DatastoreService datastore;
+  private WorkspaceFactory workspaceFactory;
+  private TaskSchedulerFactory taskSchedulerFactory;
+  @VisibleForTesting protected String QUEUE_NAME;
+
+  @Override
+  public void init(ServletConfig config) throws ServletException {
+    try {
+      authInstance = FirebaseAuth.getInstance(FirebaseAppManager.getApp());
+      taskSchedulerFactory = TaskSchedulerFactory.getInstance();
+      workspaceFactory = WorkspaceFactory.getInstance();
+      QUEUE_NAME = System.getenv("EXECUTION_QUEUE_ID");
+    } catch (IOException e) {
+      throw new ServletException(e);
+    }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    System.setProperty(
+        DatastoreServiceConfig.DATASTORE_EMPTY_LIST_SUPPORT, Boolean.TRUE.toString());
+
+    try {
+      String classCode = request.getParameter("classCode").trim();
+
+      // Retrive class entity
+      Key classKey = KeyFactory.stringToKey(classCode);
+      Entity classEntity = datastore.get(classKey);
+
+      // Delete Class entity
+      EmbeddedEntity beingHelped = (EmbeddedEntity) classEntity.getProperty("beingHelped");
+      List<EmbeddedEntity> queue = (List<EmbeddedEntity>) classEntity.getProperty("studentQueue");
+      datastore.delete(classKey);
+
+      // Delete Visit entities
+      Filter classVisitFilter = new FilterPredicate("classKey", FilterOperator.EQUAL, classKey);
+      Query visitQuery = new Query("Visit").setFilter(classVisitFilter);
+
+      for (Entity elem : datastore.prepare(visitQuery.setKeysOnly()).asIterable()) {
+        datastore.delete(elem.getKey());
+      }
+
+      // Delete Wait entities
+      Filter classWaitFilter = new FilterPredicate("classKey", FilterOperator.EQUAL, classKey);
+      Query waitQuery = new Query("Wait").setFilter(classWaitFilter);
+
+      for (Entity elem : datastore.prepare(waitQuery.setKeysOnly()).asIterable()) {
+        datastore.delete(elem.getKey());
+      }
+
+      // Delete class key from every user
+      Filter registeredClassesFilter =
+          new FilterPredicate("registeredClasses", FilterOperator.IN, Arrays.asList(classKey));
```

Comment:
```suggestion
          new FilterPredicate("registeredClasses", FilterOperator.EQUAL, classKey);
```

`FilterOperator.IN` is used to check if a property is one of many possible values. What we want to do here is the opposite. We want to check if a list contains exactly one value. For that the Equal operator is used.

Reply:
I think we want to check if the list contains the value, not if it contains just the value. If a student is registered in multiple classes alongside the one we're looking for, we'd still want them in the query.

Reply:
Yes this is what the `EQUAL` operator does! `EQUALS` is used when the property in datastore is a list. `IN` is used when you have a sing value in the datastore and you want to check if it matches any value in a list you provide to the filter.